MigrationScript/AmsOperationsHelper.ps1

```diff
@@ -0,0 +1,294 @@
+		Error = $GetProviderErrorMsg
+	}
+}
+
+<#
+.SYNOPSIS
+Function to get the managed keyvault for ams v2 monitor.
+
+.PARAMETER subscriptionId
+SubscriptionId of the AMS v2 Monitor.
+
+.PARAMETER resourceGroup
+Resource Group Name of the AMS v2 Monitor.
+
+.PARAMETER monitorName
+AMS v2 Monitor Name.
+
+.PARAMETER logger
+logger object. 
+
+.EXAMPLE
+$logger = New-Object ConsoleLogger
+GetAmsV2ManagedKv -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -logger $logger;
+#>
+function GetAmsV2ManagedKv([string]$subscriptionId, [string]$resourceGroup, [string]$monitorName, $logger)
+{    
+	[string]$managedKvName = "";
+	try
+    {
+        $response = GetAmsV2MonitorProperties -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -logger $logger
+		$managedRgName = $response.Response.properties.managedResourceGroupConfiguration.name;
+		$logger.LogInfo("Managed RG Name associated with Monitor : $monitorName is $managedRgName");
+
+		$kvDetails = Get-AzResource -ResourceGroupName $managedRgName -ResourceType Microsoft.KeyVault/vaults
+		$logger.LogInfo("Managed KV Name associated with Monitor : $monitorName is $($kvDetails.Name)");
+		$managedKvName = $kvDetails.Name;
+    }
+    catch
+    {
+        $GetProviderErrorMsg = $_.ErrorDetails.ToString();
+		$logger.LogError("GetAmsV2ManagedKv : Failed with error: ($($GetProviderErrorMsg.error.code)))", "500", "");
+    }
+
+	return @{
+		Response = $managedKvName
+		Error = $GetProviderErrorMsg
+	}
+}
+
+<#
+.SYNOPSIS
+Function to get the managed Function app name for ams v2 monitor.
+
+.PARAMETER subscriptionId
+SubscriptionId of the AMS v2 Monitor.
+
+.PARAMETER resourceGroup
+Resource Group Name of the AMS v2 Monitor.
+
+.PARAMETER monitorName
+AMS v2 Monitor Name.
+
+.PARAMETER providerType
+Provider Type.
+
+.PARAMETER logger
+logger object. 
+
+.EXAMPLE
+$logger = New-Object ConsoleLogger
+GetAmsV2ManagedFunc -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -logger $logger;
+#>
+function GetAmsV2ManagedFunc([string]$subscriptionId, [string]$resourceGroup, [string]$monitorName, [string]$providerType, $logger)
+{    
+	[string]$managedFuncName = "";
+	try
+    {
+        $response = GetAmsV2MonitorProperties -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -logger $logger
+		$managedRgName = $response.Response.properties.managedResourceGroupConfiguration.name;
+		$logger.LogInfo("Managed RG Name associated with Monitor : $monitorName is $managedRgName");
+
+		$funcDetails = Get-AzResource -ResourceGroupName $managedRgName -Name "$providerName*" -ResourceType Microsoft.Web/sites
+		$logger.LogInfo("Function app for Provider $providerName with Monitor : $monitorName is $($funcDetails.Name)");
+		$managedFuncName = $funcDetails.Name;
+    }
+    catch
+    {
+        $GetProviderErrorMsg = $_.ErrorDetails.ToString();
```

Comment:
I guess the pattern you are trying to go with is return the error value and the response. So we will check if error is present then to do some other stuff ?

Reply:
yes Amit